core/sequencer/trillian_client.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sequencer
+
+import (
+	"context"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/google/keytransparency/core/directory"
+
+	tpb "github.com/google/trillian"
+	tclient "github.com/google/trillian/client"
+)
+
+// Trillian contains Trillian gRPC clients and metadata about them.
+type Trillian struct {
+	directories directory.Storage
+	logAdmin    tpb.TrillianAdminClient
+	mapAdmin    tpb.TrillianAdminClient
+	tmap        tpb.TrillianMapClient
+	tlog        tpb.TrillianLogClient
+}
+
+// MapClient returns a verifying MapClient
+func (t *Trillian) MapClient(ctx context.Context, dirID string) (*MapClient, error) {
+	directory, err := t.directories.Read(ctx, dirID, false)
+	if err != nil {
+		glog.Errorf("adminstorage.Read(%v): %v", dirID, err)
+		return nil, status.Errorf(codes.Internal, "Cannot fetch directory info for %v", dirID)
+	}
+	mapTree, err := t.mapAdmin.GetTree(ctx, &tpb.GetTreeRequest{TreeId: directory.MapID})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot fetch map info for %v: %v", dirID, err)
+	}
+
+	c, err := tclient.NewMapClientFromTree(t.tmap, mapTree)
+	if err != nil {
+		return nil, err
+	}
+	return &MapClient{
+		MapClient: c,
+	}, nil
+}
+
+// LogClient returns a verifying LogClient.
+func (t *Trillian) LogClient(ctx context.Context, dirID string) (*tclient.LogClient, error) {
+	directory, err := t.directories.Read(ctx, dirID, false)
+	if err != nil {
+		glog.Errorf("adminstorage.Read(%v): %v", dirID, err)
+		return nil, status.Errorf(codes.Internal, "Cannot fetch directory info for %v", dirID)
+	}
+	logTree, err := t.logAdmin.GetTree(ctx, &tpb.GetTreeRequest{TreeId: directory.LogID})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot fetch log info for %v: %v", dirID, err)
+	}
+
+	// Create verifying log and map clients.
```

Comment:
Please remove maps from this sentence. Maybe even drop the comment, as the return line is self-describing?